### DISCUSSION TITLE
Trigger not started after deployment (case B02)

### DISCUSSION THREAD

**Discussion by TybulOnAzure** [Azure-Player/azure.datafactory.tools#369] [Q&A] [answered]
Hi @NowinskiK 

I'm having some issues with starting a trigger. Let me describe my case:
1. I use `1.7.000` version of azure.datafactory.tools.
2. I have a simple CI/CD pipeline that deploys ADF using selective approach. One of objects selected for deployment is my trigger (`FINES_AND_FEES_MW_US_TRIGGER`).
3. Publish options are as follows:
![image](https://github.com/Azure-Player/azure.datafactory.tools/assets/139237411/2b0025fb-510f-473a-9e05-b3635073ed1d)

4. Current state of `FINES_AND_FEES_MW_US_TRIGGER` trigger is as follows:
    - `Started` - in code repository,
    - `Stopped` - in ADF live service.
5. So I guess it falls under `B02` case from your table:
![image](https://github.com/Azure-Player/azure.datafactory.tools/assets/139237411/eadb1998-3a55-413e-9826-4a830524980e)

7. During `Stopping triggers...` step, `FINES_AND_FEES_MW_US_TRIGGER` trigger is not stopped as it was already stopped in target ADF service - that is fine:
![image](https://github.com/Azure-Player/azure.datafactory.tools/assets/139237411/ac15ac6a-d1c6-491f-8de1-1c1d71e0e705)

8. Then, during `Deployment of all ADF objects...` step, this trigger is deployed - that is fine:
![image](https://github.com/Azure-Player/azure.datafactory.tools/assets/139237411/e6922aa4-bdb0-422f-a4e4-263b6f53f6a8)

10. However, during `Starting triggers...` step, it is **not** started as adf.tools believes that it is already started which is not true:
![image](https://github.com/Azure-Player/azure.datafactory.tools/assets/139237411/18b1beef-cbe1-4611-a5e8-adb77aaed50d)

11. As a result, the trigger is in `Stopped` state in target ADF service, which is against `B02` case.

Could you please take a look at it and let me know if my understanding is correct? 

I might be wrong but I think that the issue might be in https://github.com/Azure-Player/azure.datafactory.tools/blob/master/private/Start-Triggers.ps1 file:
![image](https://github.com/Azure-Player/azure.datafactory.tools/assets/139237411/2cdfdc45-0e47-4ec9-b1ee-3eef5dcadbaa)
as maybe the code checks for disabled triggers in the JSON code instead of live ADF service.

Piotr

**Comment by TybulOnAzure** [comment]
On the other hand, `B01` case works just fine (using the same CI/CD pipeline):
1. The current state and desired state are set to `Started`, all other options are the same as previously.
2. Stopping triggers:
![image](https://github.com/Azure-Player/azure.datafactory.tools/assets/139237411/fd736ba0-a504-40cb-80c3-057702a30962)

3. Starting triggers:
![image](https://github.com/Azure-Player/azure.datafactory.tools/assets/139237411/9bc86f74-434b-446b-ab5d-974ae52fed87)

**Reply by chalabaj** [reply]
Had similar issue once there was timeout during trigger deployment. Next redeployment showed that the trigger is already started, but in  ADF live mode trigger was stopped.

**Reply by NowinskiK** [reply, marked as answer]
Thanks @TybulOnAzure & @chalabaj for details. I found the bug and finally had some time to fix it (during the flight to MVPSummit).
Follow this ticket: #386

**Reply by NowinskiK** [reply]
Guys, @TybulOnAzure & @chalabaj - this issue is fixed now and released in [1.10.0-preview](https://www.powershellgallery.com/packages/azure.datafactory.tools/1.10.0-preview).
Thanks for your patience! Please test the new version before public release.

**Reply by chalabaj** [reply]
Hi, I would like to but I cant until https://github.com/Azure-Player/azure.datafactory.tools/issues/374 is fixed as well.  Anyway. Thanks for this fix.

**Reply by NowinskiK** [reply]
@chalabaj #374 is a public preview now - no excuses against testing :)
